### ISSUE TITLE
[TEALIUM SWIFT BUG]: Runtime crash on launch with Xcode 15 beta

### ISSUE THREAD

**Issue by JhonnyBillM** [Tealium/tealium-swift#307]
**Your Environment**
iOS Version(s): **iOS 17 Beta 4 (21A5291h)**
Xcode version: **Xcode 15.0 beta 5 (15A5209g)**
Swift version: **Swift 5.0**
Tealium Swift library version: **2.10.0**
Dependency manager: **CocoaPods** 
Dependency manager version: **1.11.3**

**Describe the bug**
Unable to open iOS application when running in Xcode 15 beta due to the following runtime crash:
```
objc[69148]: Swift class extensions and categories on Swift classes are not allowed to have +load methods
```

**Does a workaround exist?**
N/A

**To Reproduce**
Run any project in Xcode 15 beta with `tealium-swift` as a dependency. 

1. `git clone https://github.com/Tealium/tealium-swift`
2. Go to one of the samples `cd tealium-swift/samples/TealiumSwiftExample`
3. Install dependencies `pod install`
4. Open `TealiumSwiftExample.xcworkspace` **with Xcode 15 beta**.
5. Run the app.
6. See runtime crash.

**Expected behavior**
No runtime crash should occur when running a project with `tealium-swift` as a dependency in Xcode 15 beta.

**Screenshots**
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://github.com/Tealium/tealium-swift/assets/18747072/644eae3b-f031-478a-a1f2-d61433076a69">
  <img alt="Screenshot of a Tealium-swift runtime crash in Xcode 15 beta 5 indicating: "objc[69148]: Swift class extensions and categories on Swift classes are not allowed to have +load methods"" src="https://github.com/Tealium/tealium-swift/assets/18747072/02b3ab9d-4d9f-47ce-9661-f3c54d741bfc">
</picture>

**Additional context**
The issue seems to be in: https://github.com/Tealium/tealium-swift/blob/15f883500c0acc6b88441176dd0c9b4cd94d4da6/tealium/core/objc/TealiumDelegateProxy%2BSwizzle.m#L26-L32 
I think we should either declare this class on the Objc side, or use a Swift based static initialization method (static variable with a closure). What do you folks think?

**Comment by craigrouse** [comment]
@JhonnyBillM Apologies for not coming back to you on this. We're taking a look at it. We've also been running the beta and haven't noticed this issue yet, which seems a little strange, but we've just updated to the latest and hopefully we can reproduce it. Thanks for the suggestions.

**Comment by Enricoza** [comment]
Hi @JhonnyBillM, thanks again for opening this. 
We just released version `2.10.1` which should solve this issue.

I'm going to close this for now, but feel free to re-open if you are still having any issues.
